src/plugins/intel_gpu/src/graph/impls/onednn/utils.cpp

```diff
@@ -318,10 +318,59 @@ dnnl::memory::desc layout_to_memory_desc(cldnn::layout l, dnnl::memory::format_t
     }
 
     dnnl::memory::data_type dt = convert_data_type(l.data_type);
-    dnnl::memory::format_tag fmt = target_fmt == dnnl::memory::format_tag::undef ? convert_data_format(l.format) : target_fmt;
-    dnnl::memory::desc res(dims, dt, fmt);
-
-    return res;
+    // If weights_layout has data_padding, add strides to memory desc
+    // Do not apply strides for blocked format cases.
+    // When strides are inserted into the memory descriptor for blocked format data, oneDNN interprets it as planar format.
+    // For blocked format, oneDNN calculates memory location based on the format information without using strides.
+    if (has_strides
+        && format::is_default_format(l.format)
```

Comment:
Shouldn't it be assert message for `has_strides && !format::is_default_format(l.format)` case?

Reply:
Remove other conditions and modify the code to determine whether or not to add strides in the memory descriptor based on the use_strides input parameter. This is necessary because, even though the same format and data type are used, there are different use cases for oneDNN kernel types. For example, matmul uses strides when dealing with i4/u4 data types, but convolution does not use strides for input or weight memory descriptors.